cmd/collectors.go

```diff
@@ -64,7 +64,11 @@ func parseCollector(s string) (t, arg string) {
 	}
 }
 
-func newCollector(collectorName, arg string, src *loader.SourceData, conf Config) (lib.Collector, error) {
+//TODO: totally refactor this...
+//nolint:funlen
+func newCollector(
+	collectorName, arg string, src *loader.SourceData, conf Config, executionPlan []lib.ExecutionStep,
+) (lib.Collector, error) {
 	getCollector := func() (lib.Collector, error) {
```

Comment:
A straightforward way to avoid the `funlen` warning would be to extract this lambda to a standalone function. :)

I'm also a fan of lambdas, but they're best used for quick and short functions (not necessarilly strictly one-liners as in Python's crippled implementation...). For example, you might want to put the `envconfig.Process(...)` call in a lambda (in that new standalone function) to avoid that specific repetition and reduce the line count a bit.

Reply:
I didn't do that since we have to fix it in `master` - after all, `newCollector` has nothing to do with the new executors, so this should be resolved by https://github.com/loadimpact/k6/pull/1202
The `envconfig` calls here are slightly different (:man_facepalming: ), so I didn't take them out in a separate lambda after all.